src/main/scala/uk/gov/nationalarchives/omega/api/services/ApiService.scala

```diff
@@ -0,0 +1,174 @@
+import uk.gov.nationalarchives.omega.api.connectors.JmsConnector
+import uk.gov.nationalarchives.omega.api.services.LocalMessage.createLocalMessage
+import uk.gov.nationalarchives.omega.api.services.ServiceState.{ Started, Starting, Stopped, Stopping }
+
+import java.nio.file.{ Files, Paths }
+import java.util.UUID
+import java.util.concurrent.atomic.AtomicReference
+
+class ApiService(val config: ServiceConfig) {
+
+  implicit val loggerFactory: LoggerFactory[IO] = Slf4jFactory[IO]
+  implicit val logger: SelfAwareStructuredLogger[IO] = LoggerFactory[IO].getLogger
+
+  private val state = new AtomicReference[ServiceState](Stopped)
+
+  def start: IO[ExitCode] =
+    // check if we can start the service
+    if (!switchState(Stopped, Starting)(config.errorOnInvalidServiceState)) {
+      IO.pure(ExitCode.Error)
+    } else {
+      attemptStartUp()
+    }
+
+  private def attemptStartUp(): IO[ExitCode] =
+    try {
+      val ec = doStart(config)
+      // switch to the started state
+      switchState(Starting, Started)(config.errorOnInvalidServiceState)
+      ec
+    } catch {
+      case e: Exception =>
+        // we are in some inconsistent state as startup failed, so we must attempt shutdown
+        try {
+          doStop()
+          switchState(Starting, Stopped)(config.errorOnInvalidServiceState)
+          IO.pure(ExitCode.Error)
+        } finally
+          logger.error(e.getMessage)
+    }
+
+  @throws[IllegalStateException]
+  def stop(): IO[Unit] = {
+    // check if we can stop the service
+    if (switchState(Started, Stopping)(config.errorOnInvalidServiceState)) {
+      logger.info("Closing connection..")
+      if (switchState(Stopping, Stopped)(config.errorOnInvalidServiceState)) {
+        doStop()
+      }
+    }
+    IO.unit // Explicitly return unit
+  }
+
+  @throws[IllegalStateException]
+  private def switchState(from: ServiceState, to: ServiceState)(errorOnInvalidServiceState: Boolean): Boolean = {
+    val switched = state.compareAndSet(from, to)
+    if (!switched) {
+      val msg = s"Unable to switch ApiService state from: $from, to: $to."
+      if (errorOnInvalidServiceState) {
+        throw new IllegalStateException(msg)
+      } else {
+        logger.warn(msg)
+      }
+    }
+    switched
+  }
+
+  private def doStart(config: ServiceConfig): IO[ExitCode] = {
+
+    // TODO(AR) - one client, how to ack a consumer message after local persistence and then process it, and then produce a response
+    // TODO(AR) how to wire up queues and services using a config file or DSL?
+
+    // TODO(AR) request queue will typically be 1 (plus maybe a few more for expedited ops), response queues will be per external application
+
+    val localQueue: IO[Queue[IO, LocalMessage]] = Queue.bounded[IO, LocalMessage](config.maxLocalQueueSize)
+    val jmsConnector = new JmsConnector(config)
+    val messageStoreFolder = Paths.get(config.tempMessageDir)
+    Files.createDirectories(messageStoreFolder)
```

Comment:
If this statement throws an exception, then the application startup fails silently. For example,
run `touch temp-messages` in the application root directory, then attempt to run the application. No exception is throw, neither does the exception get logged in the console.

Reply:
This has been refactored into a separate `getLocalMessageStore` method which should handle exceptions better.